controllers/spacebindingrequest/spacebindingrequest_controller_test.go

```diff
@@ -79,6 +124,518 @@ func TestCreateSpaceBindingRequest(t *testing.T) {
+		spacebindingrequesttest.WithSpaceRole("admin"),
+		spacebindingrequesttest.WithDeletionTimestamp(), // spaceBindingRequest was deleted
+		spacebindingrequesttest.WithFinalizer())         // has finalizer still
+	t.Run("success", func(t *testing.T) {
+		t.Run("spaceBindingRequest should be in terminating while spacebinding is deleted", func(t *testing.T) {
+			// given
+			member1 := NewMemberClusterWithClient(test.NewFakeClient(t, sbr, sbrNamespace), "member-1", corev1.ConditionTrue)
+			spaceBinding := spacebindingtest.NewSpaceBinding(janeMur.Name, janeSpace.Name, "maintainer", sbr.Name, spacebindingtest.WithSpaceBindingRequest(sbr)) // jane has maintainer, but SBR has admin
+			hostClient := test.NewFakeClient(t, base1nsTier, spaceBinding, janeSpace, janeMur)
+			ctrl := newReconciler(t, hostClient, member1)
+
+			// when
+			_, err := ctrl.Reconcile(context.TODO(), requestFor(sbr))
+
+			// then
+			require.NoError(t, err)
+			// spacebindingrequest exists with expected config and finalizer
+			spacebindingrequesttest.AssertThatSpaceBindingRequest(t, sbr.GetNamespace(), sbr.GetName(), member1.Client).
+				HasSpecSpaceRole(sbr.Spec.SpaceRole).
+				HasSpecMasterUserRecord(janeMur.Name).
+				HasConditions(spacebindingrequesttestcommon.Terminating()).
+				HasFinalizer()
+			// spacebinding was deleted
+			spacebindingtest.AssertThatSpaceBinding(t, test.HostOperatorNs, janeMur.Name, janeSpace.Name, hostClient).
+				DoesNotExist()
+
+			t.Run("spaceBindingRequest is deleted when spacebinding is gone", func(t *testing.T) {
+				// when
+				_, err := ctrl.Reconcile(context.TODO(), requestFor(sbr))
+
+				// then
+				// spaceBindingRequest was deleted
+				require.NoError(t, err)
+				spacebindingrequesttest.AssertThatSpaceBindingRequest(t, sbr.GetNamespace(), sbr.GetName(), member1.Client).DoesNotExist() // spaceBindingRequest is gone
+			})
+		})
+
+		t.Run("finalizer was already removed", func(t *testing.T) {
+			// given
+			// spaceBindingRequest has no finalizer
+			sbrNoFinalizer := spacebindingrequesttest.NewSpaceBindingRequest("lana", "lana-tenant",
+				spacebindingrequesttest.WithMUR("lana"),
+				spacebindingrequesttest.WithSpaceRole("admin"),
+				spacebindingrequesttest.WithDeletionTimestamp()) // spaceBindingRequest was deleted
+			member1 := NewMemberClusterWithClient(test.NewFakeClient(t, sbrNoFinalizer, sbrNamespace), "member-1", corev1.ConditionTrue)
+			hostClient := test.NewFakeClient(t, base1nsTier, janeSpace, janeMur)
+			ctrl := newReconciler(t, hostClient, member1)
+
+			// when
+			_, err := ctrl.Reconcile(context.TODO(), requestFor(sbrNoFinalizer))
+
+			// then
+			require.NoError(t, err)
+			spacebindingrequesttest.AssertThatSpaceBindingRequest(t, sbrNoFinalizer.GetNamespace(), sbrNoFinalizer.GetName(), member1.Client).HasNoFinalizers() // spaceBindingRequest is gone
+		})
+	})
+
+	t.Run("failure", func(t *testing.T) {
+		t.Run("SpaceBinding resource is already being deleted", func(t *testing.T) {
```

Comment:
let's make clear why it's a failure case:
```suggestion
		t.Run("SpaceBinding resource is already being deleted for more than 2 minutes", func(t *testing.T) {
```

Reply:
Done in https://github.com/codeready-toolchain/host-operator/commit/0d24598c352faf223469ee1b913d9a6fc3993772

Thanks!